src/sentry/integrations/slack/webhooks/action.py

```diff
@@ -402,10 +433,15 @@ def open_resolve_dialog(self, slack_request: SlackActionRequest, group: Group) -
         modal_payload = self.build_resolve_modal_payload(callback_id, metadata=metadata)
         slack_client = SlackSdkClient(integration_id=slack_request.integration.id)
         try:
-            slack_client.views_open(
-                trigger_id=slack_request.data["trigger_id"],
-                view=modal_payload,
-            )
+            # We need to use the action_ts as the external_id to update the modal
+            # We passed this in control when we sent the loading modal to beat the 3 second timeout
+            external_id = slack_request.data.get("actions")[0].get("action_ts")
```

Comment:
i think you might want to set some defaults in case `actions` doesn't exist? is this guaranteed to be in the first entry of the `actions` array?

Reply:
+1 to this. I'd like to factor out the `[0]`'s in general, maybe make a helper function for unpacking stuff from the request data. Type checks would also be helpful, so that we get a good exception if something isn't a sequence or mapping when we expect.

When we expect a sequence to have exactly one element, it's better to communicate/assert that with `(thing,) = sequence` rather than `sequence[0]`. If we're expecting multiple elements but only care about the first, we could clarify that as well.

Reply:
i factored out the parsing and adding more type checking and assertions